packages/plugins/test/mocha/mock.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  BuildInfo,
+  ConfigArtifacts,
+  ExecutionMode,
+  GetConfigArtifacts,
+  ParsedConfig,
+  isLiveNetwork,
+} from '@sphinx-labs/core'
+import sinon from 'sinon'
+import { Operation } from '@sphinx-labs/contracts'
+
+import { readContractArtifact } from '../../src/foundry/utils'
+import { propose } from '../../src/cli/propose'
+import { deploy } from '../../src/cli/deploy'
+import { makeSphinxContext } from '../../src/cli/context'
+
+/**
+ * Make a mocked `SphinxContext` object. Use this function if it's safe to assume that all of
+ * `SphinxContext` member functions are mocked. In integration tests, use the
+ * `makeMockSphinxContextForIntegrationTests` function instead.
+ */
+export const makeMockSphinxContext = (
```

Comment:
I'm confused why there's two functions here, when we are only using one of them. 

It seems like you want to introduce a new pattern where we have this function for unit tests, then we use the `makeMockSphinxContextForIntegrationTests` which we use for larger integration tests? That seems reasonable to me, but I wouldn't recommend making this type of change unless you're going to fully execute on it. I.e add a few new tests that use the new pattern, update the organization of the tests files to clearly differentiate between integration tests and this new type of tests, etc. It's just easy to make this type of change and then never actually come back and finish it which can leave you with a bunch of half finished work that confuses other devs. 

I don't think it's necessary to revert this since it seems like a reasonable change. I would just add documentation that this function isn't actually used currently, and provide an explanation for the intended use case. In the future, I would recommend making this type of change in it's own PR and taking a bit more time to fully think through the change and make sure that the PR includes the full pattern.

Reply:
> I'm confused why there's two functions here, when we are only using one of them.

We use `makeMockSphinxContext` [here](https://github.com/sphinx-labs/sphinx/pull/1322/files#diff-af73bf84e06d033e941580bf28757ba49a6cd79dc942568d357a5f77d0e3bfadR181) to make the `SphinxContext` for integration tests. You're right that we don't use it directly in our tests though. 

> I would provide an explanation for the intended use case

I already included an explanation:
```
/**
 * Make a mocked `SphinxContext` object. Use this function if it's safe to assume that all of
 * `SphinxContext` member functions are mocked. In integration tests, use the
 * `makeMockSphinxContextForIntegrationTests` function instead.
 */
```

Are you looking for something more specific?

Reply:
I just would like more detail on exactly what the intended use case for this function is. I don't really understand what you mean by "if it's safe to assume that all of `SphinxContext` member functions are mocked". In what case would that be exactly? 

Is this function intended to be used when testing the deploy/propose tasks themselves, or should it be used only when testing lower-level functions? What specific use case did you have in mind when you created it?